### ISSUE TITLE
Properties declared as `InitVar` are recognized as valid properties in `dataclass` classes

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#2359]
**Describe the bug**
Given the following code:

```python
from __future__ import annotations

from collections.abc import Iterable
from dataclasses import InitVar, dataclass, field
from typing import Any


@dataclass
class Container:
    iterable: InitVar[Iterable[Any]]
    items: list[Any] = field(init=False)

    def __post_init__(self, iterable: Iterable[Any]) -> None:
        self.items = list(iterable)


c = Container((1, 2, 3, 4))
c.items  # list[Any]
c.iterable  # Iterable[Any]
```

The property `c.iterable` is incorrectly shown as `Iterable[Any]` and not flagged as an error. Running the above code will cause an `AttributeError` to be raised when the property is accessed.

**Expected behavior**
The type checker should report an error for `c.iterable` which states that it cannot be accessed (similar to other properties that do not exist).

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.172

**Comment by erictraut** [comment]
Thanks for reporting the issue. Since this is a false negative issue (a missing check), I'm going to mark it as an enhancement request rather than a bug.

**Comment by erictraut** [comment]
This will be addressed in the next release. I also updated the completion provider so it has knowledge of InitVar fields and doesn't suggest them as member access completions.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.173, which I just published. It will also be included in the next release of pylance.
